src/contracts/entrypoint.rs

```diff
@@ -0,0 +1,238 @@
+                        Ok(SimulateResult::SimulationResult(res))
+                    }
+                    entry_point_api::EntryPointAPIErrors::SimulationResultWithAggregation(res) => {
+                        Ok(SimulateResult::SimulationResultWithAggregation(res))
+                    }
+                    _ => Err(EntryPointErr::UnknownErr(format!(
+                        "Simulate validation with invalid error: {:?}",
+                        op
+                    ))),
+                })
+            }
+        }
+    }
+
+    pub async fn handle_ops<U: Into<UserOperation>>(
+        &self,
+        ops: Vec<U>,
+        beneficiary: Address,
+    ) -> Result<(), EntryPointErr> {
+        self.api
+            .handle_ops(ops.into_iter().map(|u| u.into()).collect(), beneficiary)
+            .await
+            .or_else(|e| {
+                let err_msg = e.to_string();
+                Self::deserialize_error_msg(&err_msg).and_then(|op| match op {
+                    entry_point_api::EntryPointAPIErrors::FailedOp(failed_op) => {
+                        Err(EntryPointErr::FailedOp(failed_op))
+                    }
+                    _ => Err(EntryPointErr::UnknownErr(format!(
+                        "Handle ops with invalid error: {:?}",
+                        op
+                    ))),
+                })
+            })
+    }
+
+    async fn get_sender_address<U: Into<UserOperation>>(
+        &self,
+        initcode: Bytes,
+    ) -> Result<entry_point_api::SenderAddressResult, EntryPointErr> {
+        let result = self.api.get_sender_address(initcode).await;
+
+        match result {
+            Ok(_) => Err(EntryPointErr::UnknownErr(
+                "Get sender address should expect revert".to_string(),
+            )),
+            Err(e) => {
+                let err_msg = e.to_string();
+                Self::deserialize_error_msg(&err_msg).and_then(|op| match op {
+                    entry_point_api::EntryPointAPIErrors::SenderAddressResult(res) => Ok(res),
+                    entry_point_api::EntryPointAPIErrors::FailedOp(failed_op) => {
+                        Err(EntryPointErr::FailedOp(failed_op))
+                    }
+                    _ => Err(EntryPointErr::UnknownErr(format!(
+                        "Simulate validation with invalid error: {:?}",
+                        op
+                    ))),
+                })
+            }
+        }
+    }
+
+    async fn handle_aggregated_ops<U: Into<UserOperation>>(
+        &self,
+        ops_per_aggregator: Vec<U>,
+        beneficiary: Address,
+    ) -> Result<(), EntryPointErr> {
+        todo!()
+    }
+}
+
+#[derive(Debug)]
+pub enum EntryPointErr {
+    FailedOp(entry_point_api::FailedOp),
+    NetworkErr, // TODO
+    DecodeErr(String),
+    UnknownErr(String), // describe impossible error. We should fix the codes here(or contract codes) if this occurs.
+}
+
+#[derive(Debug)]
+pub enum SimulateResult {
```

Comment:
SimulateValidationResult sounds better to me (more precise based on the SC's function name